File: namsor/csharp/src/OSEG.NamsorExamples/CasteIndianBatchExample.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text.Json;

using App.Namsor.Api;
using App.Namsor.Client;
using App.Namsor.Model;

namespace OSEG.NamsorExamples;

public class CasteIndianBatchExample
{
    public static void Run()
    {
        var config = new Configuration();
        config.ApiKey = new Dictionary<string, string> {["api_key"] = "YOUR_API_KEY"};

        var personalNames1 = new FirstLastNameGeoSubdivisionIn(
            id: "id",
            firstName: "firstName",
            lastName: "lastName",
            countryIso2: "countryIso2",
            subdivisionIso: "subdivisionIso"
        );

        var personalNames2 = new FirstLastNameGeoSubdivisionIn(
            id: "id",
            firstName: "firstName",
            lastName: "lastName",
            countryIso2: "countryIso2",
            subdivisionIso: "subdivisionIso"
        );

        var personalNames = new List<FirstLastNameGeoSubdivisionIn>
        {
            personalNames1,
            personalNames2,
        };

        var batchFirstLastNameGeoSubdivisionIn = new BatchFirstLastNameGeoSubdivisionIn(
            personalNames: personalNames
        );

        try
        {
            var response = new IndianApi(config).CasteIndianBatch(
                batchFirstLastNameGeoSubdivisionIn: batchFirstLastNameGeoSubdivisionIn
            );

            Console.WriteLine(response);
        }
        catch (ApiException e)
        {
            Console.WriteLine("Exception when calling IndianApi#CasteIndianBatch: " + e.Message);
            Console.WriteLine("Status Code: " + e.ErrorCode);
            Console.WriteLine(e.StackTrace);
        }
    }
}

File: namsor/typescript-axios/src/CasteIndianBatchExample.ts
import axios, { AxiosError } from "axios";
import * as api from "namsor_client"

const configuration = new api.Configuration({
  apiKey: "YOUR_API_KEY",
});

const personalNames1: api.FirstLastNameGeoSubdivisionIn = {
  id: "id",
  firstName: "firstName",
  lastName: "lastName",
  countryIso2: "countryIso2",
  subdivisionIso: "subdivisionIso",
};

const personalNames2: api.FirstLastNameGeoSubdivisionIn = {
  id: "id",
  firstName: "firstName",
  lastName: "lastName",
  countryIso2: "countryIso2",
  subdivisionIso: "subdivisionIso",
};

const personalNames = [
  personalNames1,
  personalNames2,
];

const batchFirstLastNameGeoSubdivisionIn: api.BatchFirstLastNameGeoSubdivisionIn = {
  personalNames: personalNames,
};

new api.IndianApi(configuration).casteIndianBatch(
  batchFirstLastNameGeoSubdivisionIn,
).then(response => {
  console.log(response.data);
}).catch((error: Error | AxiosError) => {
  console.log("Exception when calling IndianApi#casteIndianBatch:");
  axios.isAxiosError(error) ? console.log(error.message) : console.log(error);
});
